app/src/main/java/de/westnordost/streetcomplete/overlays/surface/RoadSurfaceOverlay.kt

```diff
@@ -0,0 +1,116 @@
+package de.westnordost.streetcomplete.overlays.surface
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.osm.ALL_ROADS
+import de.westnordost.streetcomplete.osm.isPrivateOnFoot
+import de.westnordost.streetcomplete.osm.surface.CyclewayFootwaySurfaces
+import de.westnordost.streetcomplete.osm.surface.CyclewayFootwaySurfacesWithNote
+import de.westnordost.streetcomplete.osm.surface.INVALID_SURFACES
+import de.westnordost.streetcomplete.osm.surface.SingleSurface
+import de.westnordost.streetcomplete.osm.surface.SingleSurfaceWithNote
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissing
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissingWithNote
+import de.westnordost.streetcomplete.osm.surface.UNDERSPECIFED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.keysToBeRemovedOnSurfaceChange
+import de.westnordost.streetcomplete.osm.surface.createSurfaceStatus
+import de.westnordost.streetcomplete.overlays.Color
+import de.westnordost.streetcomplete.overlays.Overlay
+import de.westnordost.streetcomplete.overlays.PolygonStyle
+import de.westnordost.streetcomplete.overlays.PolylineStyle
+import de.westnordost.streetcomplete.overlays.StrokeStyle
+import de.westnordost.streetcomplete.overlays.Style
+import de.westnordost.streetcomplete.quests.surface.AddPathSurface
+import de.westnordost.streetcomplete.quests.surface.AddRoadSurface
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.CAR
+
+class RoadSurfaceOverlay : Overlay {
+
+    override val title = R.string.overlay_road_surface
+    override val icon = R.drawable.ic_quest_street_surface
+    override val changesetComment = "Specify road surfaces"
+    override val wikiLink: String = "Key:surface"
+    override val achievements = listOf(CAR, BICYCLIST)
+    override val hidesQuestTypes = setOf(AddRoadSurface::class.simpleName!!, AddPathSurface::class.simpleName!!)
+
+    private val handledSurfaces = Surface.values().map { it.osmValue }.toSet() + INVALID_SURFACES
+
+    override fun getStyledElements(mapData: MapDataWithGeometry): Sequence<Pair<Element, Style>> {
+        return mapData
+            .filter("""
+                ways, relations with
```

Comment:
Why relations? Roads cannot be relations.

(Same for path overlay)

Reply:
Some pedestrian squares are valid multipolygons: https://www.openstreetmap.org/relation/152813 https://www.openstreetmap.org/relation/3278602

Reply:
ic